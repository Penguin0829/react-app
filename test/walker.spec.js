const {
  walk,
  checkIfDirectoryShouldBeIgnored,
  checkIfFileShouldBeIgnored,
} = require('../src/walker');

describe('walker', () => {
  describe('walk', () => {
    it('should return list of all files in directory', () => {
      const allFiles = walk('./test/walking_test');
      expect(allFiles).toMatchSnapshot();
    });
  });
  describe('checkIfDirectoryShouldBeIgnored', () => {
    it('should ignore node_modules', () => {
      expect(checkIfDirectoryShouldBeIgnored('node_modules/blah')).toBeTruthy();
    });
  });
  describe('checkIfFileShouldBeIgnored', () => {
    it('should accept js and jsx files', () => {
      expect(checkIfFileShouldBeIgnored('/asdasd/blah.js')).toBeFalsy();
      expect(checkIfFileShouldBeIgnored('/asdasd/blah.jsx')).toBeFalsy();
      expect(checkIfFileShouldBeIgnored('C:\\Users\\username\\Desktop\\i18ize\\captive-app\\src\\App.js')).toBeFalsy();
    });
    it('should reject all other files', () => {
      expect(checkIfFileShouldBeIgnored('not_a_virus.exe')).toBeTruthy();
    });
  });
});
