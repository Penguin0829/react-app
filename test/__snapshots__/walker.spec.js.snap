// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`walker walk should return list of all files in directory 1`] = `
Array [
  "test\\\\walking_test\\\\a.js",
  "test\\\\walking_test\\\\directory\\\\b.jsx",
]
`;
